src/Integration/Binding/BindingController.cs

```diff
@@ -128,15 +128,22 @@ private void OnBind(SonarQubeProject projectInformation)
 
         void IBindingWorkflowExecutor.BindProject(SonarQubeProject projectInformation)
         {
-            ConnectionInformation connection = this.host.VisualStateManager.GetConnectedServer(projectInformation);
-            Debug.Assert(connection != null, "Could not find a connected server for project: " + projectInformation?.Key);
+            //TODO: CM2 - choose the type of binding
+            var workflow = CreateBindingWorkflow(projectInformation);
 
-            BindingWorkflow workflowExecutor = new BindingWorkflow(this.host, connection, projectInformation);
-            IProgressEvents progressEvents = workflowExecutor.Run();
+            IProgressEvents progressEvents = workflow.Run();
             Debug.Assert(progressEvents != null, "BindingWorkflow.Run returned null");
             this.SetBindingInProgress(progressEvents, projectInformation);
         }
 
+        private IBindingWorkflow CreateBindingWorkflow(SonarQubeProject projectInformation)
+        {
+            ConnectionInformation connection = this.host.VisualStateManager.GetConnectedServer(projectInformation);
+            Debug.Assert(connection != null, "Could not find a connected server for project: " + projectInformation?.Key);
+
+            return new BindingWorkflow(this.host, connection, projectInformation);
```

Comment:
Minimal refactoring - changed the controller to use the new _IBindingWorkflow_ interface.
A later still will be to introduce a new workflow for the new connected mode, and use a factory method here to decide which workflow to use.